scripts/mongodbintegrationmvp.py

```diff
@@ -0,0 +1,230 @@
+            "snapshot_data": snapshot_data,
+            "timestamp": datetime.now().isoformat(),
+            "trial_count": (
+                len(ax_client.get_trials_data_frame())
+                if ax_client.get_trials_data_frame() is not None
+                else 0
+            ),
+        }
+
+        # Insert a new document for every snapshot (no overwrite)
+        snapshots_col.insert_one(snapshot_doc)
+
+        os.remove(temp_file)
+
+        print(f"Snapshot saved to MongoDB at {snapshot_doc['timestamp']}")
+        return True
+
+    except Exception as e:
+        print(f"Error saving snapshot: {e}")
+        return False
+
+
+def load_ax_snapshot_from_mongodb(experiment_name):
+    """Load most recent Ax client snapshot from MongoDB."""
+    try:
+        # Find the most recent snapshot
+        record = snapshots_col.find_one(
+            {"experiment_name": experiment_name},
+            sort=[("timestamp", -1)],  # Most recent first
+        )
+
+        if record:
+            # Save snapshot data to temporary file
+            temp_file = f"temp_{experiment_name}_snapshot.json"
+            with open(temp_file, "w") as f:
+                json.dump(record["snapshot_data"], f)
+
+            # Load AxClient from file
+            ax_client = AxClient.load_from_json_file(temp_file)
+
+            # Clean up temp file
+            os.remove(temp_file)
+
+            print(
+                f"Loaded snapshot from {record['timestamp']} with "
+                f"{record['trial_count']} trials"
+            )
+            return ax_client
+        else:
+            print("No existing snapshot found")
+            return None
+
+    except Exception as e:
+        print(f"Error loading snapshot: {e}")
+        return None
+
+
+# Load existing experiment or create new one
+ax_client = load_ax_snapshot_from_mongodb(obj1_name)
+
+if ax_client is None:
+    # Create new experiment
+    generation_strategy = GenerationStrategy(
+        [
+            GenerationStep(
+                model=Models.SOBOL,
+                num_trials=SOBOL_TRIALS,
+                min_trials_observed=1,
+                max_parallelism=5,
+                model_kwargs={"seed": 999},  # For reproducibility
+            ),
+            GenerationStep(
+                model=Models.GPEI,
+                num_trials=-1,
+                max_parallelism=3,
+                model_kwargs={},
+            ),
+        ]
+    )
+    ax_client = AxClient(generation_strategy=generation_strategy)
+    ax_client.create_experiment(
+        name=obj1_name, parameters=parameters, objectives=objectives
+    )
+    print(f"Created new experiment with {SOBOL_TRIALS} Sobol trials")
+
+    # Save initial snapshot
+    save_ax_snapshot_to_mongodb(ax_client, obj1_name)
+else:
+    print("Resuming existing experiment")
+
+# Get current trial count to determine how many more trials to run
+current_trials = ax_client.get_trials_data_frame()
```

Comment:
Ah, good point about needing to handle max trials in the for loop. I suppose an alternative would be to have a budget variable stored in MongoDB that gets updated, but ignore that for now. More of a note to self/musing.